src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -15,40 +15,53 @@
 logger = get_logger(__name__)
 
 
-def _upload_bitstream(
+def handle_media_info(
     resource: XMLResource,
+    media_previously_uploaded: bool,
     sipi_server: Sipi,
     imgdir: str,
     permissions_lookup: dict[str, Permissions],
-) -> BitstreamInfo | None:
+) -> tuple[bool, None | BitstreamInfo]:
     """
-    This function uploads a specified bitstream file to SIPI and then returns the file information from SIPI.
+    This function checks if a resource has a bitstream
 
     Args:
-        resource: resource with that has a bitstream
+        resource: resource holding the bitstream
+        media_previously_uploaded: True if the image is already in SIPI
         sipi_server: server to upload
-        imgdir: directory of the file
         permissions_lookup: dictionary that contains the permission name as string and the corresponding Python object
+        imgdir: directory of the file
 
     Returns:
-        The information from sipi which is needed to establish a link from the resource
+        If the handling of the bitstream object was successful,
+        in case of a bitstream the object or None in case of unsuccessful try.
     """
-    pth = resource.bitstream.value  # type: ignore[union-attr]
-    img: Optional[dict[Any, Any]] = try_network_action(
-        sipi_server.upload_bitstream,
-        filepath=str(Path(imgdir) / Path(pth)),
-    )
-    internal_file_name_bitstream = img["uploadedFiles"][0]["internalFilename"]  # type: ignore[index]
-    return resource.get_bitstream_information(
-        internal_file_name_bitstream=internal_file_name_bitstream,
-        permissions_lookup=permissions_lookup,
-    )
+    bitstream = cast(XMLBitstream, resource.bitstream)
```

Comment:
I'm sceptical about this cast: with this, you erase the type information that this may be None. Afterwards you correctly check for this case, so currently it's not broken. But if in the future we delete the line with this check, then the type system does not warn us that we may have None at hand when we assume an `XMLBitstream`. Is it really needed? And if so, why?

Reply:
I had to cast it for mypy. I could not find another solution apart from ignoring it.

Reply:
that seems strange to me, because it the type of `bitstream` is `XMLBitstream | None` then mypy should understand the following match-case to infer the correct type in each branch. Where do you get errors if you remove the cast?

Reply:
![Screenshot 2023-12-11 at 15 55 04](https://github.com/dasch-swiss/dsp-tools/assets/103038637/4198dac0-891b-4df1-825e-9e879aaeaefc)

Reply:
would it work if you changed the last case to wildcard (`case _: ...`)?